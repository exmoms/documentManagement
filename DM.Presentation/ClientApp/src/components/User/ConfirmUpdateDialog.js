import React, { Component } from "react";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import { withTranslation } from "react-i18next";

class ConfirmUpdateDialog extends Component {
  constructor(props) {
    super(props);
    this.handleConfirmUpdate = this.props.handleConfirmUpdate;
    this.handleIgnoreUpdate = this.props.handleIgnoreUpdate;
    this.handleClose = this.props.handleClose;
  }
  t = this.props["t"];
  render() {
    return (
      <div>
        <Dialog
          open={this.props.open}
          aria-labelledby="alert-dialog-title"
          aria-describedby="alert-dialog-description"
        >
          <DialogTitle id="alert-dialog-title">
            {this.t("alert_exiteupdating")}
          </DialogTitle>
          <DialogContent>
            <DialogContentText id="alert-dialog-description">
              {this.t("dialog_documentchanges")}
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button
              id="Back"
              onClick={this.handleClose}
              color="primary"
              autoFocus
            >
              {this.t("Button.back")}
            </Button>
            <Button
              id="Ignore"
              onClick={this.handleIgnoreUpdate}
              color="primary"
            >
              {this.t("button_ignore")}
            </Button>
            <Button
              id="Confirm"
              type="submit"
              form="update-form"
              color="primary"
              autoFocus
            >
              {this.t("button_confirm")}
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

export default withTranslation()(ConfirmUpdateDialog);
