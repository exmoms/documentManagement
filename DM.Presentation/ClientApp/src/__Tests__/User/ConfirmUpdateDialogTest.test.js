import React from "react";
import ReactDom from "react-dom";
import ConfirmUpdateDialog from "../../components/User/ConfirmUpdateDialog";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";

afterEach(cleanup);
describe("ConfirmUpdateDialog", () => {
  test("ConfirmUpdateDialog renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<ConfirmUpdateDialog />, div);
  });

  test("ConfirmUpdateDialog matches snapshot", () => {
    const component = renderer.create(<ConfirmUpdateDialog />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("ConfirmUpdateDialog Functionality ", () => {
    let open = true;
    let handleIgnoreUpdateMock = jest.fn();
    let handleCloseMock = jest.fn();
    let handleConfirmUpdateMock = jest.fn();
    let wrapper = mount(
      <ConfirmUpdateDialog
        open={open}
        handleIgnoreUpdate={handleIgnoreUpdateMock}
        handleClose={handleCloseMock}
      />
    );
    act(() => {
      wrapper.find("#Back").at(0).props().onClick();
    });
    act(() => {
      wrapper.find("#Ignore").at(0).props().onClick();
    });
    expect(handleIgnoreUpdateMock).toHaveBeenCalledTimes(1);
  });
});
