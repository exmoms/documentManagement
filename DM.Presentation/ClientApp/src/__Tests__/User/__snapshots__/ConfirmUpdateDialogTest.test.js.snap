// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmUpdateDialog ConfirmUpdateDialog matches snapshot 1`] = `<div />`;
